packages/microsoft_exchange_server/data_stream/messagetracking/elasticsearch/ingest_pipeline/default.yml

```
@@ -54,26 +54,27 @@ processors:
 - set:
     field: email.local_id
     copy_from: microsoft.exchange.networkmessageid
-- append:
+    ignore_empty_value: true
+- set:
     field: email.sender.address
-    value: "{{{microsoft.exchange.senderaddress}}}"
-    ignore_failure: true
+    copy_from: microsoft.exchange.senderaddress
+    ignore_empty_value: true
 - dissect:
     field: microsoft.exchange.customdata
     pattern: "%{}S:OriginalFromAddress=%{_tmp.email.from.address};"
     ignore_failure: true
 - append:
     field: email.from.address
     value: "{{{_tmp.email.from.address}}}"
-    if: ctx._tmp?.email?.from?.address != null && ctx?._tmp?.email?.from?.address != "<>"
-    ignore_failure: true
+    if: ctx.microsoft?.exchange?.senderaddress != null && ctx.microsoft.exchange.senderaddress != ""
```

The `if` condition should be validating the source value being reference from the mustache template.

```suggestion
    if: ctx._tmp?.email?.from?.address != null && ctx._tmp.email.from.address != "<>"
```